src/lib/operators/aggregate.cpp

```diff
@@ -74,6 +74,12 @@ std::shared_ptr<AbstractOperator> Aggregate::_on_recreate(
   return std::make_shared<Aggregate>(recreated_input_left, _aggregates, _groupby_column_ids);
 }
 
+void Aggregate::_on_cleanup() {
+  _output.reset();
```

Comment:
Having to remember to call this in every `_on_cleanup` override is error prone. 
Proposal: Introduce `AbstractOperator::_cleanup()` which clears the output and then calls `virtual AbstractOperator::_on_cleanup()` where derived operators can do their own cleanup.

Reply:
Wait, isn't `_on_cleanup()` called after the every execution of the operator? How can you clean the output there without affecting the successor-operator? This is a bug in the current implementation, right?

The fact that the added tests didn't catch this suggests that there should be more, imo, but that's up to you

Reply:
Argh, I have just taken Aggregate's members. We have `std::shared_ptr<Table> Aggregate::_output;` and `std::shared_ptr<const Table> AbstractOperator::_output;`. WTF?

Fixed.